src/org/hammerhead226/sharkmacro/actions/ActionList.java

```diff
@@ -97,43 +109,34 @@ public boolean isFinished() {
 		 */
 		public void run() {
 			int i = 0;
-			while (i < actionList.size()) {
-				if (actionList.get(i).getStartTime() <= timer.get()) {
-					actionList.get(i).start();
-					actionList.remove(i);
+			while (i < this.al.actionList.size()) {
+				if (this.al.actionList.get(i).getStartTime() <= timer.get()) {
+					this.al.actionList.get(i).start();
+					this.al.actionList.remove(i);
 				} else {
 					i++;
 				}
 			}
-			if (actionList.size() == 0) {
-				isFinished = true;
-				thread.stop();
+			if (this.al.actionList.size() == 0) {
+				this.al.isFinished = true;
+				this.al.thread.stop();
 				return;
 			}
 		}
 	}
 
 	/**
-	 * Object that takes a runnable class and starts a new thread to call its
-	 * {@link java.lang.Runnable#run() run()} method periodically.
-	 */
-	Notifier thread;
-
-	/**
-	 * This method clones this class and all of its members.
+	 * String representation of each {@link Action} contained in this {@link ActionList}.
 	 * 
-	 * @return a new copy of this {@link ActionList}
+	 * @return the string representation
 	 */
-	protected Object clone() {
-		try {
-			ActionList al = (ActionList) super.clone();
-			al.actionList = (ArrayList<Action>) this.actionList.clone();
-			al.timer = new Timer();
-			return al;
-		} catch (CloneNotSupportedException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+	public String toString() {
+		StringBuilder sb = new StringBuilder();
+		for (Action a : actionList) {
+			sb.append(a.toString());
+			sb.append("\n");
+			sb.append("\n");
```

Comment:
FWIW personally wouldn't do `\n\n` for the print, would let code outside this do the extra.